### DISCUSSION TITLE
Easy way to linearize ast to markdown?

### DISCUSSION THREAD

**Discussion by wdscxsj** [yuin/goldmark#265] [Ideas]
Hello, first of all thanks for this wonderful project! It really saves a lot of work. The ast interface allows flexible manipulation of the nodes, but is there an easy way to linearize the tree (or any `Node`, preferably) back to markdown? Manually adding markups is boring if not error prone. Maybe another built-in renderer, or something like `go/printer`? Thanks!

**Comment by zcharym** [comment]
Hoping for the same feature. I want to append a dynamic content (e.g. api table list) into markdown file, but found it can't be reversed : (

**Comment by iwahbe** [comment]
You can use a different renderer that targets markdown. I have been using https://github.com/teekennedy/goldmark-markdown.
